src/parser.ts

```diff
@@ -0,0 +1,45 @@
+interface Token {
+  type: string;
```

Comment:
I guess this should maybe be an enum.

Reply:
In Typescript you can specifically say that a property will be a specific string:

`type: 'EXAMPLE' | 'DIFFERENT' | null`

For example.